### PR TITLE
feat(infra): Add support for rendering PersistentVolumes

### PR DESCRIPTION
**Pull request by peturgq**
# Add support for rendering PersistentVolumes
Relevant [issue](https://github.com/island-is/infrastructure/issues/1313)

## What
Make changes to DSL to allow for PV's in deployments.

## Why
In case stateful applications need to persist files in EFS or other.

### From feature deployment: 
feature-deploy-ids-efs-1

feature-deploy-ids-efs-2

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: infra/src/dsl/types/input-types.ts [hunk 1]
```diff
@@ -11,6 +11,12 @@ export interface Service {
 export type Hash = { [name: string]: Hash | string | number }
 export type ValueSource = string | ((e: Context) => string)
 export type ValueType = MissingSettingType | ValueSource
+// See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes for more info
+export type AccessModes =
+  | 'ReadWriteOnce'
+  | 'ReadWriteMany'
+  | 'ReadWriteOncePod'
+  | 'ReadOnlyMany'
```
Comment (pshomov) [inline infra/src/dsl/types/input-types.ts:19]: Maybe boil down to what we know will be used right now and in more "dev-terms" (avoid Kubernetes speak)

Comment (pshomov) [reply infra/src/dsl/types/input-types.ts:19]: Filter out the modes we will not be using (at least for now)

**Review by pshomov — CHANGES_REQUESTED** [review]
Excellent, small improvement needed.

File: infra/src/dsl/types/input-types.ts (hunk 1 above)
Comment (peturgq) [reply infra/src/dsl/types/input-types.ts]: Solved in [602dec1](https://github.com/island-is/island.is/pull/8589/commits/602dec105362ab4194924e03fbccf6a5d5b98589)

**Review by brynjarorng — APPROVED** [review]
LGTM

**Review by pshomov — APPROVED** [review]
LGTM

**Pull request merged**